base/src/com/google/idea/blaze/base/sync/aspects/BlazeIdeInterfaceAspectsImpl.java

```diff
@@ -763,6 +772,36 @@ private static BlazeBuildOutputs runBuildForTargets(
           builder, outputGroups, activeLanguages, onlyDirectDeps);
 
       return invoker.getCommandRunner().run(project, builder, buildResultHelper, context, ImmutableMap.of());
+    } finally {
+      if (!Registry.is("bazel.sync.keep.target.files")) {
+          try {
+              Files.deleteIfExists(targetPatternFile);
+          } catch (IOException e) {
+              logger.error("Failed to delete target pattern file", e);
+          }
+      }
+    }
+  }
+
+  private static @NotNull Path prepareTargetPatternFile(List<? extends TargetExpression> targets) throws BuildException {
+    Path targetPatternFile = null;
+    try {
+      targetPatternFile = Files.createTempFile("targets-", "");
```

Comment:
can we create the file in a directory under the project similar to here: https://github.com/tpasternak/bazel-intellij/blob/769b653fa5ca7a71f92c44f897584e1d3b2011de/base/src/com/google/idea/blaze/base/dependencies/BlazeQuerySourceToTargetProvider.java#L229?